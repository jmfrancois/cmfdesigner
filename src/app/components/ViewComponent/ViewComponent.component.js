import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect, Inject } from '@talend/react-cmf';

function ViewComponent(props) {
	return (
		<div>
			<ul>
				<li>Type: {props.component.type}</li>
				{props.component.hasPropTypes && <li>has PropTypes</li>}
				{props.component.displayName && <li>displayName: {props.component.displayName}</li>}
			</ul>
			<Inject component="ViewPropTypes" data={props.component.propTypes} />
		</div>
	);
}

ViewComponent.propTypes = {
	component: PropTypes.shape({
		type: PropTypes.string.isRequired,
		hasPropTypes: PropTypes.bool,
		propTypes: PropTypes.object,
		displayName: PropTypes.string,
	}),
	...cmfConnect.propTypes,
};
ViewComponent.ACTION_TYPE_DELETE_BTN = 'VIEW_COMPONENT_DELETE_BTN_CLICKED';
ViewComponent.displayName = 'ViewComponent';
ViewComponent.defaultProps = {
	component: {},
};
export default cmfConnect({
	defaultProps: {
		componentExpression: 'service#designer.components:getCurrent',
	},
})(ViewComponent);
