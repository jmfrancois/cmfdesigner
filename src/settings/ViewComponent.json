{
  "props": {
    "ViewComponent#default": {
      "itemExpression": "service#designer.components:getComponent",
      "renderIfExpression": {
        "id": "cmf.router.matchPath",
        "args": [
            {
                "path": "/components/view",
                "isExact": true
            }
        ]
      }
    }
  }
}
