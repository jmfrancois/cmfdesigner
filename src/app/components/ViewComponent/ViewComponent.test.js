import React from 'react';
import { shallow } from 'enzyme';

import ViewComponent from './ViewComponent.component';

/* global jest */

jest.mock('../../experimental-cmf/modules', () => ({
	get: () => ({
		inComponent: () => ({
			delete: jest.fn(),
		}),
	}),
}));

describe('ViewComponent', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewComponent.WrappedComponent />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
