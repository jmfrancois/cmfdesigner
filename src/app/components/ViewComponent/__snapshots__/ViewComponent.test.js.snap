// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewComponent should render 1`] = `
<div>
  <button
    className="btn btn-danger pull-right"
    onClick={[Function]}
  >
    Delete
  </button>
  <h1 />
  <Inject
    component="Files"
    componentId="component"
  />
</div>
`;
